### DISCUSSION TITLE
Accessing mounted LB3 Models and their datasources in LB4

### DISCUSSION THREAD

**Discussion by kvnwong0210** [loopbackio/loopback-next#9396] [LB3 Migration] [answered]
I'm currently running a middleware that in lb3 would be passed the app as a whole and could access the models with the method app.models.MODEL and then run the built-in create method to create a new document in the DB datasource set. I've attempted to recreate this with a lb4 server with the old lb3 mounted but the app that I'm passing in doesn't have access to the .models method.

Below is a code snipped of how I am passing in the app in the old lb3 app and how it is being used.

```
var app = (module.exports = loopback());
var model = app.models.User
model.create({"username: "John"})
```

And here is how I am trying it in the lb4 server.ts file

```
export class ExpressServer {
  public readonly app: express.Application;
  public readonly lbApp: lb3Application;
  private server?: http.Server;
  public url: String;

  constructor(options: ApplicationConfig = {}) {
    this.app = express();
    this.lbApp = new lb3Application(options);

   var model = this.lbApp.models.User
   model.create({"username: "John"})
}
```

However the lbApp doesn't have the .models method. Any idea how I can access the mounted models in Lb4?

**Comment by nflaig** [comment, marked as answer]
you can import the lb3 app, like this `const lb3app = require("../../lb3app/core/app");`, depends on your project structure

and then you can just do `lb3app.models.User`

**Reply by kvnwong0210** [reply]
Thank you. The app as per the old loopback3 directory structure was in `var lb3app = require("../../../lb3app/server/server.js")`
and from there I could run `lb3app.models.User.create({})`
